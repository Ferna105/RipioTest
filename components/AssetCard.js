import {View, Text, Image, TouchableOpacity, Alert} from 'react-native';
import * as Clipboard from 'expo-clipboard';
import { useSelector } from 'react-redux';

export default AssetCard = () => {
    const reducer = useSelector(state => state.reducer);

    const copyToClipboard = () => {
        Clipboard.setString(reducer.user_account.address);
        Alert.alert("Copiado", "La dirección fue copiada al portapapeles");
    };

    return (
        <View style={styles.card}>
            <View style={{ flexDirection: "row", justifyContent: 'space-between' }}>
                <View>
                    <Text style={styles.subtitle}>{reducer.user_account.balance} BTC</Text>
                    <Text style={styles.text}>({reducer.user_account.balance * reducer.ars_cotizations.sell} ARS)</Text>
                </View>
                <Image style={{ width: 50, height: 50 }} source={require('../assets/images/btc.png')} />
            </View>
            <TouchableOpacity activeOpacity={.5} onPress={copyToClipboard} >
                <>
                    <Text style={styles.text}>Dirección:</Text>
                    <Text style={styles.subtitle} >{reducer.user_account.address}</Text>
                </>
            </TouchableOpacity>
        </View>
    )
}