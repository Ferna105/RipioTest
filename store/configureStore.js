import { createStore, combineReducers, applyMiddleware } from 'redux';
import reducer from '../reducers';
import thunk from 'redux-thunk'

const rootReducer = combineReducers(
    { reducer: reducer },
);
const configureStore = () => {
    return createStore(rootReducer, applyMiddleware(thunk));
}
export default configureStore;