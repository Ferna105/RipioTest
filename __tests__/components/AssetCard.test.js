import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../store/configureStore';
const store = configureStore()

import AssetCard from '../../components/AssetCard';
it('renders correctly', () => {
    const tree = renderer.create(
        <Provider store = { store }>
            <AssetCard />    
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});