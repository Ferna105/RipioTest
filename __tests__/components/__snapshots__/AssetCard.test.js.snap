// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f9f9f9",
      "borderRadius": 10,
      "marginVertical": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#841584",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
         BTC
      </Text>
      <Text
        style={
          Object {
            "color": "#841584",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
        (
        NaN
         ARS)
      </Text>
    </View>
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#841584",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Direcci√≥n:
    </Text>
    <Text
      style={
        Object {
          "color": "#841584",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    />
  </View>
</View>
`;
